django/db/backends/postgresql/schema.py

```diff
@@ -59,7 +57,12 @@ def _create_like_index_sql(self, model, field):
         return None
 
     def _alter_column_type_sql(self, model, old_field, new_field, new_type):
-        """Make ALTER TYPE with SERIAL make sense."""
+        # Add casting to alter command when new_type != old_type
+        if self.connection.data_types[old_field.get_internal_type()] != \
+           self.connection.data_types[new_field.get_internal_type()]:
+            self.sql_alter_column_type = "ALTER COLUMN %(column)s TYPE %(type)s USING %(column)s::%(type)s"
```

Comment:
You want to avoid altering `self` here as subsequent calls will reuse this attribute even if this branch's conditions don't match.

Reply:
I can not understand what is your mean exactly. is it possible to explain a little bit more?
Do you mean to don't change the `self.sql_alter_column_type`?

Reply:
What I mean is that if you call this method twice on the same `SchemaEditor` instance and the `if` pass all conditions on the first time but not on the second the assignment to `self.sql_alter_column_type` will be used on the second method call's to `super()` when it shouldn't.

Reply:
Thank you. I added an else block for it.